app/grandchallenge/participants/models.py

```diff
@@ -54,3 +56,88 @@ def delete(self, *args, **kwargs):
 
     class Meta:
         unique_together = (("challenge", "user"),)
+
+
+def string_type_schema():
+    return {"type": "string"}
+
+
+class RegistrationQuestion(UUIDModel):
+    challenge = models.ForeignKey(
+        Challenge,
+        on_delete=models.CASCADE,
+    )
+
+    question_text = models.TextField(blank=False)
+    question_help_text = models.TextField(blank=True)
+
+    schema = models.JSONField(
+        default=string_type_schema,
+        help_text="A JSON schema definition against which an answer is validated. See https://json-schema.org/."
+        "Only Draft 7, 6, 4 or 3 are supported.",
+        validators=[JSONSchemaValidator()],
+    )
+
+    required = models.BooleanField(default=True)
+
+    @property
+    def has_answers(self):
+        return RegistrationQuestionAnswer.objects.filter(
+            question=self
+        ).exists()
+
+    class Meta:
+        unique_together = (("question_text", "challenge"),)
+
+    def clean(self):
+        super().clean()
+
+        if (
+            type(self)
+            .objects.filter(
+                challenge=self.challenge,
+                question_text=self.question_text,
+            )
+            .exclude(pk=self.pk)
+            .exists()
+        ):
+            raise ValidationError(
+                f"There is already an existing {self._meta.model._meta.verbose_name} with this question text"
+            )
+
+
+class RegistrationQuestionAnswer(models.Model):
+    registration_request = models.ForeignKey(
+        RegistrationRequest,
+        on_delete=models.CASCADE,
+    )
+    question = models.ForeignKey(
+        RegistrationQuestion,
+        on_delete=models.CASCADE,
+    )
+
+    answer = models.JSONField(blank=True, editable=False)
```

Comment:
Is making this non-editable correct, you want the user to provide an answer so it should be editable (e.g. usable in forms), no?

Reply:
Not sure, we have a single request per user+challenge that can either be rejected or accepted. I sort of think that it should not be editable, but I can also forsee a future where they are. There isn't any path planned in this pitch to allow for it to be editable. Might as well save us a migration down the road. I'll make it editable.

Reply:
Unresolving this as I think you misunderstand a concept in Django still. Please check the meaning of "editable" in Django https://docs.djangoproject.com/en/5.1/ref/models/fields/#editable

> If False, the field will not be displayed in the admin or any other ModelForm. They are also skipped during model validation. Default is True.

It means that you cannot use it in forms, even the creation one, so the user _would not_ be able to provide an answer.

Reply:
btw, this I think is the danger of including new models in a PR that are not fully hooked up yet.